src/components/Button/Button.tsx

```diff
@@ -57,7 +58,11 @@ export const Button = (props: TButtonPropTypes): ReactElement => {
             />
           ) : null}
           {/* {buttonSecondaryText ? <span className="btn__text">{buttonSecondaryText}</span> : null} */}
-          {buttonText ? <span className="btn__text">{buttonText}</span> : null}
+          {buttonText || children ? (
+            <span className="btn__text">
+              {buttonText || (typeof children === 'function' ? children() : children)}
```

Comment:
Why is it possible for children to be a function, but you don't pass any props directly to children?

Reply:
Just in case if Front Dev wants do some logic inside of button children, for example
`<Button>{() => {
let text = "initial text"
if (doSomeCheckings()) {
text = "another text"
} else {
text. = "mek ayl text )) "
}
}}</Button`

generally just giving full flexibility to the developer

Reply:
{items.map(item => {
   let text = 'initial text';
  if (doSomeCheckings(item)) {
     text = 'initial text';
  } else {
    text = 'mek ayl text'
  }
  return <Button>{text}</Button>
})}

i think you can do same without passing fc children,

 const Component= ({ children }) => {
  const data = { name: "asd" };

  return (
    <div>
      {children(data)}
    </div>
  );
};

const App = () => (
  <MyComponent>
    {(data) => (
      <div>
        <p>Name: {data.name}</p>
      </div>
    )}
  </MyComponent>
);

in this case its ok to have children which is function, you can pass data to children

Reply:
i fully agree with you Har jan. Even there is more ways to handle this part without using children as a function. 
Just as i said the main idea is giving **flexibility**.

Reply:
ok Raz jan, maybe in the future we will use that part and we will pass args to children

Reply:
removed Har jan ❤️